### PR TITLE
feat: add X-AR-IO-Trusted header and safely reenable ar-io-peers data source

### PR DESCRIPTION
**Pull request by djwhitt**
## Summary

- Add new `X-AR-IO-Trusted` header to indicate data source trustworthiness
- Implement peer data filtering to only accept verified or trusted content from ar-io peers
- Safely re-enable ar-io-peers as the last fallback option in on-demand data retrieval

## Background

Misconfigured peers may inadvertently serve bad content (e.g., provider default landing pages) instead of actual Arweave data. This change adds safety measures to prevent serving such content by only accepting data from peers that explicitly indicate their content is either verified or trusted.

### Core Safety Implementation
- **New Header**: Added `X-AR-IO-Trusted` header constant and response handling
- **Peer Filtering**: ar-io peers data source now validates incoming data by checking `X-AR-IO-Verified` and `X-AR-IO-Trusted` headers
- **Safety First**: Only streams data from peers that indicate content is either verified (`X-AR-IO-Verified: true`) or trusted (`X-AR-IO-Trusted: true`)
- **Error Handling**: Properly destroys streams and retries with next peer when data is rejected

### Configuration Update  
- **Re-enable Peers**: Added `ar-io-peers` to the end of `ON_DEMAND_RETRIEVAL_ORDER` as a safe fallback
- **Priority Order**: Maintains existing source priority while adding ar-io-peers as last resort:
  1. `s3` (AWS S3 data source)
  2. `trusted-gateways` (configured trusted gateways)  
  3. `chunks` (transaction chunks)
  4. `tx-data` (raw Arweave transaction data)
  5. `ar-io-peers` (AR-IO network peers with verification)

### Test Coverage
- **Comprehensive Tests**: Added 6 new test cases covering all verification scenarios
- **Edge Cases**: Tests for missing headers, retry behavior, and various header combinations
- **Backward Compatibility**: Updated existing tests to work with new filtering requirements

## Benefits

- **Content Integrity**: Prevents serving misconfigured content from peers
- **Safe Peer Network**: Enables broader participation in ar-io peer network with safety guarantees
- **Fallback Resilience**: Provides additional data source option when primary sources fail
- **Configurable**: Users can still override data source order via environment variables

## Test Plan

- [x] All existing tests continue to pass
- [x] New peer filtering tests validate verification logic  
- [x] Integration tests verify proper header setting
- [x] Linting and code quality checks pass

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/routes/data/handlers.test.ts [hunk 1]
```diff
@@ -775,6 +775,318 @@ st
...
+            .get('/not-a-real-id')
+            .set('Range', 'bytes=2-3')
+            .set('If-None-Match', '"test-hash"')
+            .expect(304)
+            .then((res: any) => {
+              assert.equal(res.headers['etag'], '"test-hash"');
+              assert.equal(res.text || '', '');
+            });
+        });
+      });
```
[the remark this answers is not in the document]
Comment (djwhitt) [reply src/routes/data/handlers.test.ts]: Sure. Why not.

File: src/routes/data/handlers.ts [hunk 2]
```diff
@@ -59,18 +72,30 @@ const setDigestStableVerifiedHeaders = ({
...
+      (data.cached || req.method === REQUEST_METHOD_HEAD)
+    ) {
       res.setHeader(headerNames.digest, dataAttributes.hash);
-      res.setHeader('ETag', dataAttributes.hash);
+      res.setHeader('ETag', `"${dataAttributes.hash}"`);
     }
   }
+
+  // Set trusted header based on data source
+  res.setHeader(headerNames.trusted, data.trusted ? 'true' : 'false');
```
[the remark this answers is not in the document]
Comment (djwhitt) [reply src/routes/data/handlers.ts:89]: Nah, this is fine.

File: src/routes/data/handlers.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (djwhitt) [reply src/routes/data/handlers.test.ts]: I'm good. I added the tests here: 1ee5e7f1a76830d43a0108de5ad320562e94d766
